src/platform_impl/macos/event_loop.rs

```diff
@@ -141,6 +141,7 @@ impl<T> EventLoop<T> {
         F: 'static + FnMut(Event<'_, T>, &RootWindowTarget<T>, &mut ControlFlow),
     {
         self.run_return(callback);
+        drop(self._callback.take());
```

Comment:
Upon further though, I think you should move this to the end of `run_return`, as the closure passed to `run_return` should also be dropped before returning.

Reply:
Donezo